ddtrace/tracer/post_processor_test.go

```diff
@@ -0,0 +1,389 @@
+		assert.Equal(true, tracer.runProcessor(spans))
+	})
+
+	t.Run("reject", func(t *testing.T) {
+		assert := assert.New(t)
+		tracer, _, _, stop := startTestTracer(t, WithPostProcessor(func([]ddtrace.ReadWriteSpan) bool { return false }))
+		defer stop()
+		spans := []*span{newBasicSpan("http.request"), newBasicSpan("db.request")}
+		assert.Equal(false, tracer.runProcessor(spans))
+	})
+
+	t.Run("reject-condition", func(t *testing.T) {
+		assert := assert.New(t)
+		tracer, _, _, stop := startTestTracer(t, WithPostProcessor(func(spans []ddtrace.ReadWriteSpan) bool {
+			for _, span := range spans {
+				if span.Tag(ext.SpanName) == "reject.request" {
+					return false
+				}
+			}
+			return true
+		}))
+		defer stop()
+		spans := []*span{newBasicSpan("http.request"), newBasicSpan("reject.request")}
+		assert.Equal(false, tracer.runProcessor(spans))
+	})
+
+	t.Run("empty-spans", func(t *testing.T) {
+		assert := assert.New(t)
+		tracer, _, _, stop := startTestTracer(t, WithPostProcessor(func(spans []ddtrace.ReadWriteSpan) bool { return true }))
+		defer stop()
+		spans := []*span{}
+		assert.Equal(true, tracer.runProcessor(spans))
+		assert.Equal(0, len(spans))
+	})
+
+	t.Run("tag", func(t *testing.T) {
+		assert := assert.New(t)
+		tracer, _, _, stop := startTestTracer(t, WithPostProcessor(func(spans []ddtrace.ReadWriteSpan) bool {
+			for _, span := range spans {
+				if span.Tag(ext.SpanName) == "http.request" {
+					span.SetTag("custom", "val")
+				}
+				if span.Tag(ext.SpanName) == "db.request" {
+					span.SetTag("metric", float64(1))
+				}
+			}
+			return true
+		}))
+		defer stop()
+		spans := []*span{newBasicSpan("http.request"), newBasicSpan("db.request")}
+		assert.Equal(true, tracer.runProcessor(spans))
+		for _, span := range spans {
+			if span.Name == "http.request" {
+				assert.Equal("val", span.Meta["custom"])
+				assert.Equal(float64(0), span.Metrics["metric"])
+			}
+			if span.Name == "db.request" {
+				assert.Equal("", span.Meta["custom"])
+				assert.Equal(float64(1), span.Metrics["metric"])
+			}
+		}
+	})
+}
+
+func runProcessorTestEndToEnd(t *testing.T, testFunc func(sls spanLists), processor func([]ddtrace.ReadWriteSpan) bool, startSpans func()) {
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer r.Body.Close()
+		var sls spanLists
+		if err := msgp.Decode(r.Body, &sls); err != nil {
+			t.Fatal(err)
+		}
+		testFunc(sls)
+	}))
+	defer srv.Close()
+
+	// Disables call to checkEndpoint which sends an empty payload.
+	os.Setenv("DD_TRACE_STARTUP_LOGS", "false")
+	defer os.Unsetenv("DD_TRACE_STARTUP_LOGS")
+	transport := newDefaultTransport()
+	httpTransport, ok := transport.(*httpTransport)
+	if !ok {
+		t.Fatal("transport type assertion failed")
+	}
+	// Setting the traceURL manually instead of WithAgentAddr because
+	// the latter would force us to have an endpoint "/v0.4/traces",
+	// which is subject to change.
+	httpTransport.traceURL = srv.URL
```

Comment:
We could also replace all occurrences of `/v0.4/traces` by a variable (eg. `tracesEndpoint`) which would remove the need to set the `traceURL` manually here. We could then have a `tracesEndpoint` endpoint in the test and set the addr via `WithAgentAddr`.